<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Transferring Data to HPC Cluster</title>

<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Manual Annotation of Functional Genes in Non-Model Species</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="HPC_organization.html">1. Command line Basics</a>
</li>
<li>
  <a href="slurm_basics.html">2. Slurm Basics</a>
</li>
<li>
  <a href="Kegg_fasta.html">3. Identify Functional Genes</a>
</li>
<li>
  <a href="Data_Download.html">4. Transfer Data to the Cluster</a>
</li>
<li>
  <a href="genome_mining.html">5. Mine the Genome using the UNIX shell</a>
</li>
<li>
  <a href="geneious_settings.html">6. Prepare Geneious Project</a>
</li>
<li>
  <a href="geneious_annoataion.html">7. Annotate the Genome using Geneious</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Transferring Data to HPC Cluster</h1>

</div>


<p>Now that we have organized directories on a cluster, we will move the required files for mining the genome to the cluster.</p>
<p><strong>Required Files</strong>:</p>
<ul>
<li><p>Genome of non-model species</p></li>
<li><p>FASTA file of functional genes of interest (created using the KEGG database)</p></li>
<li><p>RNA sequence data</p></li>
</ul>
<div id="genome-of-non-model-species" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Genome of Non-model species</h1>
<ol style="list-style-type: decimal">
<li><p>Identify the non-model species genome using the <a href="https://www.ncbi.nlm.nih.gov/datasets/taxonomy/8954/">NCBI Database</a></p>
<ul>
<li>For this example, the <a href="https://www.ncbi.nlm.nih.gov/datasets/taxonomy/8954/"><em>Falco peregrinus</em> reference genome</a> was used.</li>
</ul></li>
<li><p>Once on the reference genome page, click the ‘DOWNLOAD’ option (Figure 1.A).
On the resulting window, select three download options (Figure 1.B):</p>
<ol style="list-style-type: decimal">
<li><p>Genome Sequences (FASTA)</p></li>
<li><p>Annotation features (GTF)</p></li>
<li><p>Annotation features (GFF)</p>
<div class="figure">
<img src="Images/Slide1%203.jpg" alt="" />
<p class="caption">Figure 1. Genome file downloading options for the Falco peregrinus reference genome as of August 2023. Image A highlights where to click to download. Image B highlights required files to download.</p>
</div></li>
</ol></li>
<li><p>Move all the downloaded genome files into a project folder on the home computer.</p>
<ol style="list-style-type: decimal">
<li>These three files will be needed for the manual annotation process in Geneious after the genome has been mined, but for now the <code>.fna.gz</code> file will need to be transferred onto the computing cluster. It is recommended to use <code>scp</code> for this step.</li>
</ol></li>
<li><p>Open a new terminal shell and input the scp command.
A password will be required to access the files through the HPC before the transfer can begin.</p></li>
</ol>
<pre><code>scp &lt;user:/current/path/to/file&gt; &lt;user:/location/where/you/would/like/to/move/file/to&gt;</code></pre>
<p>In the example below, all downloaded genome files were transferred to the genome directory using the wildcard <code>*</code> option.
This option gathers all files that start with that prefix:</p>
<pre><code>scp /Users/username/Downloads/GCF_023634155.1_bFalPer1.pri_genomic.* HPC-criteria:/scratch/username/tlr_mining/p_falcon/genome</code></pre>
</div>
<div id="genes-of-interest" class="section level1" number="2">
<h1><span class="header-section-number">2</span> Genes of Interest</h1>
<ol style="list-style-type: decimal">
<li>Transfer the fasta file created from the KEGG database into the blast directory:</li>
</ol>
<pre><code>scp /Users/username/Downloads/Zebra_Finch_TLRs.fasta HPC-criteria:/scratch/username/tlr_mining/p_falcon/blast</code></pre>
<ol start="2" style="list-style-type: decimal">
<li><p>Login to the HPC and confirm files have been transferred to the appropriate directories.</p>
<p><em>Depending on the available space on your computer and institutional options, other file upload options could include <code>wget</code> and <a href="https://www.globus.org/data-transfer">Globus</a>.</em></p></li>
</ol>
</div>
<div id="rna-sequence-data" class="section level1" number="3">
<h1><span class="header-section-number">3</span> RNA sequence data</h1>
<div id="search-the-sra-database-for-rna-seq-data" class="section level2" number="3.1">
<h2><span class="header-section-number">3.1</span> Search the SRA database for RNA-seq data</h2>
<p>We will use the <a href="https://www.ncbi.nlm.nih.gov/sra">SRA database</a> to find RNA sequencing data.
These files will provide protein-level data we can use to mine for the functional genes of interest.
Since are interested in the TLRs, our preferred sample will be blood.</p>
<ol style="list-style-type: decimal">
<li>Search the scientific name of the non-model organism (<em>Falco peregrinus</em>) of interest in the search bar.</li>
<li>On the resulting page a list of submission will appear, but we want to focus on RNA source samples. On the left side of the screen select the RNA filter option under ‘Source’ (Figure 1).</li>
<li>SRA submission are not consistently labeled. Some might detail the sample type, but that is not always the case. You will want to click on different submission until you find one that from the preferred sample source.
<ul>
<li>If a sample source is not detailed in the Accession information area, you might need to deep dive into the literature to find the research paper for this sample and review the methods section for sample collection.</li>
<li>In this example, Accession: SRX9708906 (Kola7) was a blood sample and is usable for our workflow.</li>
</ul></li>
</ol>
<div class="figure">
<img src="Images/Figure_%20SRA_search.jpg" alt="" />
<p class="caption">Figure 1. SRA database search of RNA source samples for <em>F. peregrinus</em>. Top five results displayed from August 2023.</p>
</div>
<ol start="4" style="list-style-type: decimal">
<li><p>View the Kola7 SRA information page, we will want to note the run identification so that we can use SRA Toolkit to download the RNA-seq data onto the cluster (Figure 7).</p>
<ul>
<li>For the Kola7 SRA submission the run identification is SRR13279135
<ul>
<li><p>Click on the run number to learn more about the submission and data.</p>
<div class="figure">
<img src="images/Figure_SRA_details.jpg" alt="" />
<p class="caption">Figure 2. SRA database details for Kola7.</p>
</div></li>
</ul></li>
</ul></li>
</ol>
<p>Now that we know the Run ID from the SRA database, we will submit a job using SRA Toolkit to download the RNA-seq files to the cluster.
Next we will need to download the relevant SRA files.</p>
</div>
<div id="use-sra-toolkit-to-download-rna-seq-data-to-the-cluster" class="section level2" number="3.2">
<h2><span class="header-section-number">3.2</span> Use SRA Toolkit to download RNA-seq data to the cluster</h2>
<p><a href="https://github.com/ncbi/sra-tools">SRA Toolk</a>it is a module created to download SRA data.
Most data is too large to download directly from the SRA repository, so this toolkit is usually a necessary step.</p>
<ol style="list-style-type: decimal">
<li><p>Create a job file in the terminal using the <code>nano</code> command</p>
<p><code>nano sra_pfalcon</code></p></li>
<li><p>Adjust the slurm script example below and copy into the nano screen.</p></li>
<li><p>Run the slurm script, using the <code>sbatch</code> command:</p>
<p><code>sbatch sra_pfalcon</code></p></li>
<li><p>Once the job is finished running, confirm there are two files are in the sra directory.</p>
<ul>
<li>For this example, the resulting files should be SRR13279135_1.fastq and SRR13279135_2.fastq.</li>
</ul></li>
</ol>
<div id="sra-toolkit-slurm-script-example" class="section level5" number="3.2.0.0.1">
<h5><span class="header-section-number">3.2.0.0.1</span> SRA toolkit slurm script example:</h5>
<pre><code>#!/bin/bash
#SBATCH -J sra_pfalcon
#SBATCH -o log.o%j 
#SBATCH -p bsudfq
#SBATCH -N 1
#SBATCH -n 14
#SBATCH -t 14-00:00:00
#SBATCH --mail-type=END
#SBATCH --mail-type=BEGIN
#SBATCH --mail-type=FAIL
#SBATCH --mail-user=emailaddress

#########################
# Load Modules:
module load bioinformatics/2023-05-04
module load sratoolkit/2.10.9

# Sort Path:
datapath=/bsuscratch/username/tlr_mining/p_falcon/sra
## Sort Path section is required if you are creating a jobfile to submit job from a different directory than the directory the required files for the job are located.

# SRA toolkit:
fastq-dump -I --split-files SRR13279135</code></pre>
<p><strong>Note:</strong> Be mindful of module versions available on your cluster.
If you receive an Error Code 3, then the SRA file rather large you will need to use <code>prefetch</code> and <code>fasterq-dump</code>.
See the example below.</p>
<pre><code>prefetch SRR21639343 --max-size 96000000000
fasterq-dump SRR21639343 --split-files</code></pre>
</div>
<div id="explanation" class="section level5" number="3.2.0.0.2">
<h5><span class="header-section-number">3.2.0.0.2</span> Explanation:</h5>
<pre><code>- max-size: prefetch has a 20 GB maximum, so the limit will need to be manually increased for accession requests larger than 20 GB
- split-flies: sra files for the gfalcon were to big so I had to use prefetch. This required some configuration and then the sbatch file.</code></pre>
</div>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
