<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Prepare Geneious Workspace</title>

<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Manual Annotation of Functional Genes in Non-Model Species</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="slurm_basics.html">1. Slurm Basics</a>
</li>
<li>
  <a href="HPC_organization.html">2. HPC Workspace</a>
</li>
<li>
  <a href="Kegg_fasta.html">3. Identify Functional Genes</a>
</li>
<li>
  <a href="Data_Download.html">4. Transfer Data to the Cluster</a>
</li>
<li>
  <a href="genome_mining.html">5. Mine the Genome using the UNIX shell</a>
</li>
<li>
  <a href="geneious_settings.html">6. Geneious Project Preparation</a>
</li>
<li>
  <a href="geneious_annoataion.html">7. Manual Annotation</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Prepare Geneious Workspace</h1>

</div>


<div id="required-program" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Required Program</h1>
<p><a href="https://www.geneious.com/">Geneious</a> is a bioinformatics software that we will use to visualize the sequence data.
While Geneious is an incredible platform, it is expensive.
They offer a 1-month free trial and provide instructors with student access codes during semester sessions.</p>
<p>Once Geneious is downloaded on your home computer, learn about the <a href="https://manual.geneious.com/en/latest/2-GeneiousInterface.html">software interface options</a> before moving forward.</p>
</div>
<div id="required-files" class="section level1" number="2">
<h1><span class="header-section-number">2</span> Required Files</h1>
<p>For manual genome annotation, the following uncompressed files are required to be imported into Geneious:</p>
<ol style="list-style-type: decimal">
<li><p>Genome in FNA format (*.fna)</p>
<ol style="list-style-type: decimal">
<li>This file contains the alignment of nucleotides and protein sequences in a fasta file format.</li>
</ol></li>
<li><p>Genome in GFF format (*.gff)</p>
<ol style="list-style-type: decimal">
<li>This file details genomic features in a tab-delimited format. </li>
</ol></li>
<li><p>BAM files from the BLAST step of the cluster process.</p></li>
<li><p>Blast *gtf file</p>
<ol style="list-style-type: decimal">
<li>This gtf file is created from the blastn.out file. Follow the instructions on this <a href="https://docs.google.com/spreadsheets/d/1kkGccJadvwPx84QSyUvCvDTzy21k8zBX/edit?usp=share_link&amp;ouid=115937912009066843426&amp;rtpof=true&amp;sd=true">workbook</a> to create the .gft file.</li>
</ol></li>
<li><p>StringTie *.gft file</p></li>
</ol>
</div>
<div id="importing-files" class="section level1" number="3">
<h1><span class="header-section-number">3</span> Importing Files</h1>
<p>Once you have gathered the required files for Geneious you are ready to create a project folder in Geneious and import these files in the order listed above. </p>
<p>Right-click ‘Local” select the ’New Folder’ option, and name your folder based on the research project.</p>
<p>Importing files into Geneious requires some additional steps, files should be imported in your project folder in the proper order, as detailed in the steps below:</p>
<ol style="list-style-type: decimal">
<li><p>Drag your genome *fna file into the ‘Drop files here to import’ area of Geneious.</p>
<ul>
<li><p>During the import process, Geneious will prompt a Grouping Sequences option.
Select the ‘Keep sequences separate’ option.</p>
<ul>
<li>If Geneious prompts Ambiguous Sequences, select the ‘Nucleotide sequences’ option.</li>
</ul></li>
</ul></li>
<li><p>Select All imported .fna files using the checkbox above the document table.
Then drag and drop the genome *.gff file into the document table area.</p></li>
<li><p>Each BAM file must be imported onto the corresponding sequence file name. </p>
<ol style="list-style-type: decimal">
<li><p>Select a single sequence file (these files have the small DNA icon, see image below) then drag the BAM file on top of the highlighted sequencing file. </p>
<p><img src="Images/Geneious_nucleotide_Sequence_icon.png" width="171" height="32" /></p></li>
<li><p>Import BAM files one at a time. </p></li>
<li><p>You will only have BAM files for sequence files that have been identified to have the genes of interest.</p></li>
<li><p>Once the BAM file has been imported, you will see a contig file below the sequence file.
This file has an icon with three angled dashes (see image below).</p>
<p><img src="Images/Geneious_contig_icon.png" width="189" /></p></li>
<li><p>Select all files and drop in the stringtie file.</p></li>
<li><p>Select all files and drop in the blastHIT .gtf file.</p></li>
</ol></li>
</ol>
</div>
<div id="geneious-settings" class="section level1" number="4">
<h1><span class="header-section-number">4</span> Geneious Settings</h1>
<p>In this section, we will select settings used for visualization; however, there are many settings an options while working with annotations in Geneious.
Review the <a href="https://manual.geneious.com/en/latest/8-Annotations.html">Geneious User Manual</a> to determine additional preferred settings for your project.</p>
<p>To start, select a single contig file from the document table.
A contig view will appear in the Document Viewer area and to the right of the viewer is a settings panel (Figure 1).
Note the six different icons on the right side.
We will work through adjusting default settings for most options.
Do not change default settings unless indicated in the process below:</p>
<ol style="list-style-type: decimal">
<li>General: Select Complement and translation</li>
<li>Display: Unclick Concensus at the top. In the Translation area, set the Translation on Consensus &amp; Reference. Set the Translation Options Relative to Reference.</li>
<li>Graphs: Ensure Show Graphs and the Log Scale option under Coverage are selected.</li>
<li>Annotations and Tracks: In the tracks area, you will see the imported genome, StringTie, and blastnHIT files. For tracks with options, only have CDS and Exon selected.</li>
<li>Advanced: Under the Annotations section, Adjust Labels to none and select and unselect the Compress annotations option, as needed.</li>
</ol>
<p>Now you are ready to annotate this contig!</p>
<p><em>Please note that these setting adjustments will be done for each contig you are working on.</em></p>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
