---
title: "Mining genome using Linux/Unix Command Line"
output:
#  bookdown::pdf_document2:
#    toc: TRUE
  bookdown::html_document2: 
    toc: TRUE
    toc_float: TRUE
    theme: cosmo
link-citations: yes
fontsize: 12pt
editor_options: 
  markdown: 
    wrap: sentence
---
Now we are ready to mine the genome!

In this section, I have broken down each module and provided an example of the slurm script dependencies and job steps with some explanations.
Be sure to include the shebang and resource request portions of the slurm file prior to submitting a job.

To mine the genome, submit jobs for each required module in the following order.
Some modules will take longer than others to run.
Confirm each job has completed successfully before moving onto the next.

If a job fails to run, review the log file using the `nano` or `less` command.
and troubleshoot accordingly.

# Trim Galore:

TrimGalore is used to trim the Illumina adapters from the RNA-seq data.
You will need to identify the file names of the RNA-seq data downloaded from the SRA Toolkit step.

## TrimGalore slurm script example

```         
# Sort path:
datapath=/scratch/username/tlr_mining/gyrfalcon/sra

# Module load

. ~/.bashrc  # this line sources your bashrc
conda activate trimgalore-env
trim_galore --paired --illumina SRR21639343_1.fastq SRR21639343_2.fastq
```

### TrimGalore Options Explanation

# Hisat 2:

Run the [hisat2 aligner](http://daehwankimlab.github.io/hisat2/) to align the adapter-trimmed RNA-seq data to the Peregrine Falcon reference [genome](https://www.ncbi.nlm.nih.gov/datasets/taxonomy/8954/)

## hisat2 slurm script example

```         
# Load modules:

module load python/3.9.7
module load hisat2/2.2.1 
module load samtools/1.14

# Sort Paths:
HiSatSam=/scratch/username/tlr_mining/p_falcon/hisat_SAM
RNAseq=/scratch/username/tlr_mining/p_falcon/sra
genome=/scratch/username/tlr_mining/p_falcon/genome

# Hisat 2:
hisat2-build -p 20 ${genome}/GCF_023634155.1_bFalPer1.pri_genomic.fna ${genome}/bFalPer1.pri /
hisat2 -q --phred33 --no-unal -p 20 -x ${genome}/bFalPer1.pri -1 ${RNAseq}/SRR13279135_1_val_1.fq -2 ${RNAseq}/SRR13279135_2_val_2.fq -S ${HiSatSam}/bFalPer1.pri_align.sam
```

### Hisat2 Options Explanation

```         
- Phred33: quality threshold
- p: number of processors available 
- x: genome_name for the index that we just created 
- 1: read.1.fq read.2.fq (specify read files) 
- S: provide a name for an output file in *.sam format
```

# Samtools

We will use three XXX of [samtools](http://www.htslib.org/doc/samtools.html) to select the aligned reads we want to visualize on the reference.

-   **Samtools view** will keep only the reads that align well and align in their primary alignment position.

-   **Samtools sort** is used when further subsetting alignment reads by reads aligning by one particular contig or chromosome versus another.

-   **Samtools index** assists in downstream sub setting and calculating expression of the genes of interest.

## samtools slurm script example:

```         
# Load modules:
module load python/3.9.7
module load samtools/1.14

# Sort paths:
HiSatSam=/scratch/username/tlr_mining/p_falcon/hisat_SAM
samtools=/scratch/username/tlr_mining/p_falcon/samtools

# Samtools:
samtools view -b -@ 20 -f 3 -F 256 -q 40 ${HiSatSam}/bFalPer1.pri_align.sam > ${samtools}/bFalPer1.pri_align.bam 
samtools sort -@ 16 ${samtools}/bFalPer1.pri_align.bam > ${samtools}/bFalPer1.pri_align_sorted.bam 
samtools index -b ${samtools}/bFalPer1.pri_align_sorted.bam
```

### Samtools Options Explanation

```         
- b: adjust output format
- @: number of processors
- q: subsetting based on quality of alignmnet (40 is high quality map from read to genome)
- f: reads to include (3: both ends of paired end reads to map and on same scaffold)
- F: reads to exclude (256: not primary alignment - only keep primary alignment)
- output: bam file (binary alignment file required for Geneious)
```

# BLAST

## blast slurm script example:

```         
# Module load - Activate python environment
. ~/.bashrc
conda activate blast-env 
conda install -c conda-forge rb-parallel parallel

# Load Modules:
module load blast/2.11.0
module load samtools/1.14

# Sort Paths:
data=/scratch/username/tlr_mining/p_falcon/blast
genome=/scratch/username/tlr_mining/p_falcon/genome
bam=/scratch/username/tlr_mining/p_falcon/samtools

# Blast
makeblastdb -in ${genome}/GCF_023634155.1_bFalPer1.pri_genomic.fna -dbtype nucl -parse_seqids -out ${data}/bFalPer1.pri_genomeDB
tblastn -db ${data}/bFalPer1.pri_genomeDB -query ${data}/Zebra_Finch_TLRs.fasta -out ${data}/bFalPer1.pri_TLR_blastn.out -outfmt 6 -num_threads 7 -evalue 0.000001

cut -f2 ${data}/bFalPer1.pri_TLR_blastn.out | sort | uniq > ${data}/Pfalcon_TLR_contig.txt
cat ${data}/Pfalcon_TLR_contig.txt | parallel "samtools view -bh ${bam}/bFalPer1.pri_align_sorted.bam {} > {}.bam"
```

# Stringtie:

Stringtie is run on the whole genome, not the output from the samtools view step.

## stringtie slurm script example:

```         
# Load modules:
module load stringtie/2.1.4

# Sort paths
data=/scratch/username/tlr_mining/p_falcon/samtools

# Stringtie
stringtie -p 24 -o pfal_stringtie_out.gtf ${data}/bFalPer1.pri_align_sorted.bam
```

### Explanation

```         
- p: number of processors
- o: output file name option
- input file: can be SAM, BAM, or CRAN rna-seq align and sorted file. This is created in the samtools step. 
```
