<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Mine the Genome using Linux/Unix Command Line</title>

<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Manual Annotation of Functional Genes in Non-Model Species</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="slurm_basics.html">1. Slurm Basics</a>
</li>
<li>
  <a href="HPC_organization.html">2. HPC Workspace</a>
</li>
<li>
  <a href="Kegg_fasta.html">3. Identify Functional Genes</a>
</li>
<li>
  <a href="Data_Download.html">4. Transfer Data to the Cluster</a>
</li>
<li>
  <a href="genome_mining.html">5. Mine the Genome using the UNIX shell</a>
</li>
<li>
  <a href="geneious_settings.html">6. Geneious Project Preparation</a>
</li>
<li>
  <a href="geneious_annoataion.html">7. Manual Annotation</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Mine the Genome using Linux/Unix Command Line</h1>

</div>


<p><strong>Now we are ready to mine the genome!</strong></p>
<p>In this section, each step is listed in the order you should proceed.
Examples of the slurm script dependencies and job steps explanations are provided, but be sure to include the shebang and resource request portions of the slurm file prior to submitting a job.</p>
<p>Some modules will take longer than others to run.
Examples of output files are also provided, so you can confirm necessary output files are resulting from each step.
Confirm each job has completed successfully before moving onto the next.
It is recommended to review files last edited time and sizing using <code>ls -lht</code>.</p>
<p>If a job fails to run or if the file sizes are empty or suspiciously low, review the log file using the <code>nano</code> or <code>less</code> command and troubleshoot accordingly.</p>
<div id="trim-galore" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Trim Galore</h1>
<p>TrimGalore is used to trim the Illumina adapters from the RNA-seq data.
Be sure to identify the file names of the RNA-seq data downloaded from the SRA Toolkit step.
Trim Galore is run using bioconda, so activating an environment was required for this module.</p>
<div id="slurm-script-example" class="section level2" number="1.1">
<h2><span class="header-section-number">1.1</span> Slurm script example</h2>
<pre><code># Sort path:
datapath=/scratch/username/tlr_mining/gyrfalcon/sra

# Module load
. ~/.bashrc  # this line sources your bashrc
conda activate trimgalore-env
trim_galore --paired --illumina SRR21639343_1.fastq SRR21639343_2.fastq</code></pre>
<div id="dependencies-explanation" class="section level3" number="1.1.1">
<h3><span class="header-section-number">1.1.1</span> Dependencies Explanation</h3>
<pre><code>-- paired: identifies that they SRA files are paired end sequencing reads and the files are identified with the _1 and _2 at the end of the respective file names.
-- illumina: identifies that illumina sequencing adapters are present and will be trimmed from the reads.</code></pre>
<div id="output-files" class="section level4" number="1.1.1.1">
<h4><span class="header-section-number">1.1.1.1</span> Output files</h4>
<p>Four files should appear in the sra directory when this process is complete:</p>
<ul>
<li><p>Two <strong>trimming_report.txt</strong> files (one for each sra file)</p></li>
<li><p>A <strong>val_1.fq</strong> and <strong>val_2.fq</strong> file for each sra file, indicating they are the results of the adapter trimming process.</p></li>
</ul>
</div>
</div>
</div>
</div>
<div id="hisat-2" class="section level1" number="2">
<h1><span class="header-section-number">2</span> Hisat 2</h1>
<p>Run <a href="http://daehwankimlab.github.io/hisat2/">hisat2 aligner</a> to align the adapter-trimmed RNA-seq data to the reference genome of the non-model species.</p>
<div id="slurm-script-example-1" class="section level2" number="2.1">
<h2><span class="header-section-number">2.1</span> Slurm script example</h2>
<pre><code># Load modules:

module load python/3.9.7
module load hisat2/2.2.1 
module load samtools/1.14

# Sort Paths:
HiSatSam=/scratch/username/tlr_mining/p_falcon/hisat2
RNAseq=/scratch/username/tlr_mining/p_falcon/sra
genome=/scratch/username/tlr_mining/p_falcon/genome

# Hisat 2:
hisat2-build -p 20 ${genome}/GCF_023634155.1_bFalPer1.pri_genomic.fna ${genome}/bFalPer1.pri /
hisat2 -q --phred33 --no-unal -p 20 -x ${genome}/bFalPer1.pri -1 ${RNAseq}/SRR13279135_1_val_1.fq -2 ${RNAseq}/SRR13279135_2_val_2.fq -S ${HiSatSam}/bFalPer1.pri_align.sam</code></pre>
<div id="dependencies-explanation-1" class="section level3" number="2.1.1">
<h3><span class="header-section-number">2.1.1</span> Dependencies Explanation</h3>
<pre><code>- Phred33: quality threshold
- p: number of processors available 
- x: genome_name for the index that we just created 
- 1: read.1.fq read.2.fq (specify read files) 
- S: provide a name for an output file in *.sam format</code></pre>
<div id="output-file" class="section level4" number="2.1.1.1">
<h4><span class="header-section-number">2.1.1.1</span> Output file</h4>
<p>In the Hisat2 directory, one file should appear as the result and indicated in the -S option dependencies.
This file should be in <strong>.sam</strong> format.</p>
</div>
</div>
</div>
</div>
<div id="samtools" class="section level1" number="3">
<h1><span class="header-section-number">3</span> Samtools</h1>
<p>Three packages of <a href="http://www.htslib.org/doc/samtools.html">samtools</a> are used to select the aligned reads we want to visualize on the reference.</p>
<ul>
<li><p><strong>Samtools view</strong> will keep only the reads that align well and align in their primary alignment position.</p></li>
<li><p><strong>Samtools sort</strong> is used when further subsetting alignment reads by reads aligning by one particular contig or chromosome versus another.</p></li>
<li><p><strong>Samtools index</strong> assists in downstream sub setting and calculating expression of the genes of interest.</p></li>
</ul>
<div id="slurm-script-example-2" class="section level2" number="3.1">
<h2><span class="header-section-number">3.1</span> Slurm script example</h2>
<pre><code># Load modules:
module load python/3.9.7
module load samtools/1.14

# Sort paths:
HiSatSam=/scratch/username/tlr_mining/p_falcon/hisat_SAM
samtools=/scratch/username/tlr_mining/p_falcon/samtools

# Samtools:
samtools view -b -@ 20 -f 3 -F 256 -q 40 ${HiSatSam}/bFalPer1.pri_align.sam &gt; ${samtools}/bFalPer1.pri_align.bam 
samtools sort -@ 16 ${samtools}/bFalPer1.pri_align.bam &gt; ${samtools}/bFalPer1.pri_align_sorted.bam 
samtools index -b ${samtools}/bFalPer1.pri_align_sorted.bam</code></pre>
<div id="dependencies-explanation-2" class="section level3" number="3.1.1">
<h3><span class="header-section-number">3.1.1</span> Dependencies Explanation</h3>
<pre><code>- b: adjust output format
- @: number of processors
- q: subsetting based on quality of alignmnet (40 is high quality map from read to genome)
- f: reads to include (3: both ends of paired end reads to map and on same scaffold)
- F: reads to exclude (256: not primary alignment - only keep primary alignment)
- output: bam file (binary alignment file required for Geneious)</code></pre>
<div id="output-files-1" class="section level4" number="3.1.1.1">
<h4><span class="header-section-number">3.1.1.1</span> Output Files</h4>
<p>In the samtools directory, three files should appear as indicated in the output options from the slurm script.
Files should have the three following formats:</p>
<ol style="list-style-type: decimal">
<li><p>align.bam</p></li>
<li><p>align_sorted.bam</p></li>
<li><p>align_sorted.bam.bai</p></li>
</ol>
</div>
</div>
</div>
</div>
<div id="blast" class="section level1" number="4">
<h1><span class="header-section-number">4</span> BLAST</h1>
<p><a href="https://www.metagenomics.wiki/tools/blast">BLAST</a> is a software tool that allows users to search the <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi">NCBI database</a> for similar nucleotide or protein sequences.
In this section, we will complete steps:</p>
<ol style="list-style-type: decimal">
<li><p>Make a database of functional genes of interest from the reference genome of the non-model organism identified using the fasta file created of function genes of interest identified in the model organism.
Completed using <code>makeblastdb</code>.</p></li>
<li><p>Search translated nucleotide databases for the protein sequences of the function genes of interest.
Completed using <code>tblastn</code>.</p></li>
<li><p>Use the subsetted aligned reads from samtools with the database of matches identified from step 2 to identify which contigs of the reference genome of the model organism have the functional genes of interest.
For this step we will run the processing in parallel using <code>rb-parrell</code>, which requires bioconda.</p></li>
</ol>
<div id="blast-slurm-script-example" class="section level2" number="4.1">
<h2><span class="header-section-number">4.1</span> BLAST slurm script example</h2>
<pre><code># Module load - Activate python environment
. ~/.bashrc
conda activate blast-env 
conda install -c conda-forge rb-parallel parallel

# Load Modules:
module load blast/2.11.0
module load samtools/1.14

# Sort Paths:
data=/scratch/username/tlr_mining/p_falcon/blast
genome=/scratch/username/tlr_mining/p_falcon/genome
bam=/scratch/username/tlr_mining/p_falcon/samtools

# Blast
makeblastdb -in ${genome}/GCF_023634155.1_bFalPer1.pri_genomic.fna -dbtype nucl -parse_seqids -out ${data}/bFalPer1.pri_genomeDB
tblastn -db ${data}/bFalPer1.pri_genomeDB -query ${data}/Zebra_Finch_TLRs.fasta -out ${data}/bFalPer1.pri_TLR_blastn.out -outfmt 6 -num_threads 7 -evalue 0.000001

cut -f2 ${data}/bFalPer1.pri_TLR_blastn.out | sort | uniq &gt; ${data}/Pfalcon_TLR_contig.txt
cat ${data}/Pfalcon_TLR_contig.txt | parallel &quot;samtools view -bh ${bam}/bFalPer1.pri_align_sorted.bam {} &gt; {}.bam&quot;</code></pre>
<div id="blast-dependencies-explanation" class="section level3" number="4.1.1">
<h3><span class="header-section-number">4.1.1</span> BLAST Dependencies Explanation</h3>
<pre><code>Step 1: makeblastdb
- in: reference genome of non-model species
- dbtype: nucleic acid (nucl) or protein (prot) sequences specified
- out: user determined file name for the output database created

Step 2: tblastn
- db: database created from the makeblastdb job
- query: functional genes identified from the KEGG database
- out: user determined file name for the output
- outfmt 6: blast iudentified output format 6. User has option to make their own format
- num_threads: dependent upon computing capacity, but providing a value can speed up the module running process
- evalue: specifies the number of hits of similar quality. Lower evalue means less hits but higher score. Higher evalue means more hits, but lower score.

Step 3: In this step, rbparallel is used to run processes simultaneoulsy in efforts to search for the tblasn hits of TLR fasta sequences in the aligned sam files in each contig of the genome that has been identified to have a hit.</code></pre>
<div id="output-files-2" class="section level4" number="4.1.1.1">
<h4><span class="header-section-number">4.1.1.1</span> Output Files</h4>
<p>Many files will be created from this model.
Output names were identified in the slurm file, but a general list is detailed below.</p>
<ul>
<li><p>A set of nine <strong>_genomeDB</strong> files with multiple file formats will be created.</p></li>
<li><p><strong>blastn.out</strong> file contains blast output results in a tad-delinated organization. These results should align with a manual tblastn search on the NCBI database.
<em>This file will be needed for the manual annotation process in Geneious.</em></p></li>
<li><p><strong>contig.txt</strong> file contains a list of contigs that have been identified to have the functional genes of interest present.
<em>This file will be needed for the manual annotation process in Geneious.</em></p></li>
<li><p>A set of <strong>.bam</strong> files will be present.
The number of <strong>.bam</strong> files will vary, but should align with the number of contigs identified in the <strong>contig.txt</strong> file.
<em>These files will be needed for the manual annotation process in Geneious.</em></p></li>
</ul>
</div>
</div>
</div>
</div>
<div id="subsetting" class="section level1" number="5">
<h1><span class="header-section-number">5</span> Subsetting</h1>
<p>In the BLAST step, a set of <strong>.bam</strong> files were created for each contig that has been identified to contain the functional genes of interest.
Once ready, these files will be exported to the desktop and imported into Geneious.</p>
<p>Some of these files are can be incredibly large and if the home computer has limited storage and ram, subsetting the files can be helpful for saving space and computational power.
Review file sizes using <code>ls -lht</code> in the blast directory to determine which files need to be subset.</p>
<p>Samtools view will be used for subsetting and in this example, the goal file size of 250 mb.</p>
<p>In the example below, the original file was ~500 mb and the file was subset by 50% (0.05) to reduce the size to ~250 mb.
Generally, not all files require subsetting, but a loop could be created for this process as a learning challenge!</p>
<div id="slurm-script-example-3" class="section level2" number="5.1">
<h2><span class="header-section-number">5.1</span> Slurm script example</h2>
<pre><code># Load module:
module load samtools/1.14

# Sort paths:
HiSatSam=/scratch/username/tlr_mining/p_falcon/blast

#samtools
samtools view -s 0.50 -b NC_051190.1.bam &gt; NC_051190.1_0.050_subset.bam</code></pre>
<div id="dependencies-explanations" class="section level3" number="5.1.1">
<h3><span class="header-section-number">5.1.1</span> Dependencies Explanations</h3>
<pre><code>- s: is the amount to subset the bam file
- b: input bam file to be subset
- &gt;: user determined name of the output subset bam file; it is recommended to detail the amount the file was subset by.</code></pre>
<div id="output-files-3" class="section level4" number="5.1.1.1">
<h4><span class="header-section-number">5.1.1.1</span> Output Files</h4>
<p>A single subset.bam file should be created depending on the user determined output name.
Use <code>ls -lht</code> to confirm goal file size was met.</p>
</div>
</div>
</div>
</div>
<div id="stringtie" class="section level1" number="6">
<h1><span class="header-section-number">6</span> Stringtie</h1>
<p>Stringtie assembles RNA-sequencing reads to transcripts, and is run on the whole genome, not the output from the samtools view step.</p>
<div id="slurm-script-example-4" class="section level2" number="6.1">
<h2><span class="header-section-number">6.1</span> slurm script example</h2>
<pre><code># Load modules:
module load stringtie/2.1.4

# Sort paths
data=/scratch/username/tlr_mining/p_falcon/samtools

# Stringtie
stringtie -p 24 -o pfal_stringtie_out.gtf ${data}/bFalPer1.pri_align_sorted.bam</code></pre>
<div id="dependencies-explanation-3" class="section level3" number="6.1.1">
<h3><span class="header-section-number">6.1.1</span> Dependencies Explanation</h3>
<pre><code>- p: number of processors
- o: output file name option
- input file: can be SAM, BAM, or CRAN rna-seq align and sorted file. This is created in the samtools step. </code></pre>
<div id="output-file-1" class="section level4" number="6.1.1.1">
<h4><span class="header-section-number">6.1.1.1</span> Output File</h4>
<p>A file with the ending <strong>stringtie_out.gtf</strong> will be present in the samtools directory.
<em>This file will be needed for the manual annotation process in Geneious.</em></p>
</div>
</div>
</div>
</div>
<div id="secure-copy-files" class="section level1" number="7">
<h1><span class="header-section-number">7</span> Secure Copy Files</h1>
<p>Now the genome has been mined and resulting files need to be securely transferred from the cluster to the home computer.
For this step, <code>scp</code> is recommended. Review the data download section for a <code>scp</code> use refresher.</p>
<p>The following files are to be transferred to the home computer:</p>
<ul>
<li><p>stringtie directory</p>
<ul>
<li>stringtie_out.gtf file</li>
</ul></li>
<li><p>blast directory</p>
<ul>
<li><p>All the .bam files.
If files were subset, do not download the original file, just the subset file.</p></li>
<li><p>contig.txt file</p></li>
<li><p>blastn.out file</p></li>
</ul></li>
</ul>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
