<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Mining genome using Linux/Unix Command Line</title>

<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Manual Annotation of Functional Genes in Non-Model Organisms</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="Getting_Started.html">Scan Raw Data</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Mining genome using Linux/Unix Command Line</h1>

</div>


<p>Now we are ready to mine the genome!</p>
<p>In this section, I have broken down each module and provided an example of the slurm script dependencies and job steps with some explanations.
Be sure to include the shebang and resource request portions of the slurm file prior to submitting a job.</p>
<p>To mine the genome, submit jobs for each required module in the following order.
Some modules will take longer than others to run.
Confirm each job has completed successfully before moving onto the next.</p>
<p>If a job fails to run, review the log file using the <code>nano</code> or <code>less</code> command.
and troubleshoot accordingly.</p>
<div id="trim-galore" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Trim Galore:</h1>
<p>TrimGalore is used to trim the Illumina adapters from the RNA-seq data.
You will need to identify the file names of the RNA-seq data downloaded from the SRA Toolkit step.</p>
<div id="trimgalore-slurm-script-example" class="section level2" number="1.1">
<h2><span class="header-section-number">1.1</span> TrimGalore slurm script example</h2>
<pre><code># Sort path:
datapath=/scratch/username/tlr_mining/gyrfalcon/sra

# Module load

. ~/.bashrc  # this line sources your bashrc
conda activate trimgalore-env
trim_galore --paired --illumina SRR21639343_1.fastq SRR21639343_2.fastq</code></pre>
<div id="trimgalore-options-explanation" class="section level3" number="1.1.1">
<h3><span class="header-section-number">1.1.1</span> TrimGalore Options Explanation</h3>
</div>
</div>
</div>
<div id="hisat-2" class="section level1" number="2">
<h1><span class="header-section-number">2</span> Hisat 2:</h1>
<p>Run the <a href="http://daehwankimlab.github.io/hisat2/">hisat2 aligner</a> to align the adapter-trimmed RNA-seq data to the Peregrine Falcon reference <a href="https://www.ncbi.nlm.nih.gov/datasets/taxonomy/8954/">genome</a></p>
<div id="hisat2-slurm-script-example" class="section level2" number="2.1">
<h2><span class="header-section-number">2.1</span> hisat2 slurm script example</h2>
<pre><code># Load modules:

module load python/3.9.7
module load hisat2/2.2.1 
module load samtools/1.14

# Sort Paths:
HiSatSam=/scratch/username/tlr_mining/p_falcon/hisat_SAM
RNAseq=/scratch/username/tlr_mining/p_falcon/sra
genome=/scratch/username/tlr_mining/p_falcon/genome

# Hisat 2:
hisat2-build -p 20 ${genome}/GCF_023634155.1_bFalPer1.pri_genomic.fna ${genome}/bFalPer1.pri /
hisat2 -q --phred33 --no-unal -p 20 -x ${genome}/bFalPer1.pri -1 ${RNAseq}/SRR13279135_1_val_1.fq -2 ${RNAseq}/SRR13279135_2_val_2.fq -S ${HiSatSam}/bFalPer1.pri_align.sam</code></pre>
<div id="hisat2-options-explanation" class="section level3" number="2.1.1">
<h3><span class="header-section-number">2.1.1</span> Hisat2 Options Explanation</h3>
<pre><code>- Phred33: quality threshold
- p: number of processors available 
- x: genome_name for the index that we just created 
- 1: read.1.fq read.2.fq (specify read files) 
- S: provide a name for an output file in *.sam format</code></pre>
</div>
</div>
</div>
<div id="samtools" class="section level1" number="3">
<h1><span class="header-section-number">3</span> Samtools</h1>
<p>We will use three XXX of <a href="http://www.htslib.org/doc/samtools.html">samtools</a> to select the aligned reads we want to visualize on the reference.</p>
<ul>
<li><p><strong>Samtools view</strong> will keep only the reads that align well and align in their primary alignment position.</p></li>
<li><p><strong>Samtools sort</strong> is used when further subsetting alignment reads by reads aligning by one particular contig or chromosome versus another.</p></li>
<li><p><strong>Samtools index</strong> assists in downstream sub setting and calculating expression of the genes of interest.</p></li>
</ul>
<div id="samtools-slurm-script-example" class="section level2" number="3.1">
<h2><span class="header-section-number">3.1</span> samtools slurm script example:</h2>
<pre><code># Load modules:
module load python/3.9.7
module load samtools/1.14

# Sort paths:
HiSatSam=/scratch/username/tlr_mining/p_falcon/hisat_SAM
samtools=/scratch/username/tlr_mining/p_falcon/samtools

# Samtools:
samtools view -b -@ 20 -f 3 -F 256 -q 40 ${HiSatSam}/bFalPer1.pri_align.sam &gt; ${samtools}/bFalPer1.pri_align.bam 
samtools sort -@ 16 ${samtools}/bFalPer1.pri_align.bam &gt; ${samtools}/bFalPer1.pri_align_sorted.bam 
samtools index -b ${samtools}/bFalPer1.pri_align_sorted.bam</code></pre>
<div id="samtools-options-explanation" class="section level3" number="3.1.1">
<h3><span class="header-section-number">3.1.1</span> Samtools Options Explanation</h3>
<pre><code>- b: adjust output format
- @: number of processors
- q: subsetting based on quality of alignmnet (40 is high quality map from read to genome)
- f: reads to include (3: both ends of paired end reads to map and on same scaffold)
- F: reads to exclude (256: not primary alignment - only keep primary alignment)
- output: bam file (binary alignment file required for Geneious)</code></pre>
</div>
</div>
</div>
<div id="blast" class="section level1" number="4">
<h1><span class="header-section-number">4</span> BLAST</h1>
<div id="blast-slurm-script-example" class="section level2" number="4.1">
<h2><span class="header-section-number">4.1</span> blast slurm script example:</h2>
<pre><code># Module load - Activate python environment
. ~/.bashrc
conda activate blast-env 
conda install -c conda-forge rb-parallel parallel

# Load Modules:
module load blast/2.11.0
module load samtools/1.14

# Sort Paths:
data=/scratch/username/tlr_mining/p_falcon/blast
genome=/scratch/username/tlr_mining/p_falcon/genome
bam=/scratch/username/tlr_mining/p_falcon/samtools

# Blast
makeblastdb -in ${genome}/GCF_023634155.1_bFalPer1.pri_genomic.fna -dbtype nucl -parse_seqids -out ${data}/bFalPer1.pri_genomeDB
tblastn -db ${data}/bFalPer1.pri_genomeDB -query ${data}/Zebra_Finch_TLRs.fasta -out ${data}/bFalPer1.pri_TLR_blastn.out -outfmt 6 -num_threads 7 -evalue 0.000001

cut -f2 ${data}/bFalPer1.pri_TLR_blastn.out | sort | uniq &gt; ${data}/Pfalcon_TLR_contig.txt
cat ${data}/Pfalcon_TLR_contig.txt | parallel &quot;samtools view -bh ${bam}/bFalPer1.pri_align_sorted.bam {} &gt; {}.bam&quot;</code></pre>
</div>
</div>
<div id="stringtie" class="section level1" number="5">
<h1><span class="header-section-number">5</span> Stringtie:</h1>
<p>Stringtie is run on the whole genome, not the output from the samtools view step.</p>
<div id="stringtie-slurm-script-example" class="section level2" number="5.1">
<h2><span class="header-section-number">5.1</span> stringtie slurm script example:</h2>
<pre><code># Load modules:
module load stringtie/2.1.4

# Sort paths
data=/scratch/username/tlr_mining/p_falcon/samtools

# Stringtie
stringtie -p 24 -o pfal_stringtie_out.gtf ${data}/bFalPer1.pri_align_sorted.bam</code></pre>
<div id="explanation" class="section level3" number="5.1.1">
<h3><span class="header-section-number">5.1.1</span> Explanation</h3>
<pre><code>- p: number of processors
- o: output file name option
- input file: can be SAM, BAM, or CRAN rna-seq align and sorted file. This is created in the samtools step. </code></pre>
</div>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
