---
title: "Manual Annotation of Targeted Functional Genes in Non-model Species"
output:
#  bookdown::pdf_document2:
#    toc: TRUE
  bookdown::html_document2: 
    toc: TRUE
    toc_float: TRUE
    theme: cosmo
link-citations: yes
fontsize: 12pt
editor_options: 
  markdown: 
    wrap: sentence
---

```{js logo-js, echo=FALSE}
$(document).ready(function() {
});
```

```{r packages, echo=FALSE, warning=FALSE, include=FALSE}
library(knitr)
library(rmarkdown)
library(bookdown)
#library(distill)
library(knitcitations)
library(formatR)
library(devtools)
library(kfigr)
library(dplyr)
library(kableExtra)
library(tufte)
#Generate BibTex citation file for all R packages used to produce report
knitr::write_bib(.packages(), file = 'packages.bib')
```

```{r repo, echo=FALSE, fig.align = 'center', out.width = '100%'}
knitr::include_graphics("Images/GEM3_nov5_logo.png")
```

# Research Background
Over the last 10 years, affordable access to sequencing platforms has expanded the availability of quality genomic datasets in publicly available repositories. The challenge is no longer sequencing, but the manual and computational effort to sift through these repositories to mine and analyze these genomes. 

This workflow aims to provide a template for researchers to manually annotate genes of interest from organisms with available genomic data. To do this, we aim to walk readers through the process with a specific example: the manual annotation of toll-like receptors (TLRs) between Falco peregrinus and Falco rusticolus.

# Vision Statement

This genome annotation workflow was created by Idaho EPSCoR [GEM3](https://www.idahogem3.org) participants.
The purpose of this workflow module is to share resources and processes for genome annotation using publicly available data.

# What Will Be Covered?

Please follow this workflow to learn about how to manually annotate functional genes of interest from non-model organisms.

**Disclaimer: This workflow was developed in August 2023 with bioinformatic and genomic research recommendations at that time. Please note that program versions and recommendations will likely change with time due to the nature of this quickly evolving field.**

# What Is Required?

You will need access to an High Performance Computing (HPC) machine, [Geneious software](https://www.geneious.com/), as well as R for downstream analyses. Knowledge of job script format, how to submit jobs, and how to compute in an interactive node specific to your institution's HPC will also be necessary.

# Acknowledgements

# Licensing

<a rel="license" href="http://creativecommons.org/licenses/by/4.0/"><img alt="Creative Commons License" style="border-width:0" src="https://i.creativecommons.org/l/by/4.0/88x31.png" /></a><br />This work is licensed under a <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
